org-code-javabuilder/theater/src/main/java/org/code/theater/Stage.java

```diff
@@ -420,18 +421,19 @@ public void play() {
       this.progressPublisher.onPlay(this.audioWriter.getTotalAudioLength());
       this.gifWriter.writeToGif(this.image, 0);
       this.audioWriter.writeToAudioStream();
-      // We must call close before write so that the streams are flushed.
-      this.close();
+      // We must call onExecutionEnded() before write so that the streams are flushed.
+      this.onExecutionEnded();
```

Comment:
Maybe a nit (or misunderstanding on my part), but could we rename the contents of `onExecutionEnded()` into a helper that gets called both here and in `onExecutionEnded()`?

Reply:
Yep sure